### PR TITLE
Document using Azure ACR Helm repositories

### LINKED ISSUE #1726: Azure ACR Helm repository auth doesn't work
PR #1671 allows you to mount a `repositories.yaml` file inside the flux helm operator container in order to authenticate with private repositories.  It appears to only 'sort of' work though.

After preparing and mounting my repositories file and deploy flux, I see this in the flux-helm-operator logs:

```
ts=2019-02-11T17:00:35.327615047Z caller=chartsync.go:343 component=chartsync info="chart download failed" releaseName=acr-java resource=acr-java:helmrelease/acr-java err="Looks like \"https://hmcts.azurecr.io/helm/v1/repo/\" is not a valid chart repository or cannot be reached: Failed to fetch https://hmcts.azurecr.io/helm/v1/repo/index.yaml : 401 Unauthorized"
```

However, I also see this in the same logs (edited for brevity):

```
ts=2019-02-11T16:40:56.961576439Z caller=chartsync.go:333 component=chartsync warning=...Successfully got an update from the \"hmcts\" chart repository\n..."
```

So it is able to update charts from the repository apparently.

Furthermore, if I `kubectl exec` into the flux-helm-operator pod and run `helm install` on the chart from the private repo, it works perfectly fine - the chart installs.

This is using an Azure ACR Helm repository.  Open to any suggestions on how to troubleshoot this further if anyone has any...

### LINKED ISSUE #1733: Document how to prepare a repositories.yaml for Azure ACR Helm repo
Much of this is explained in #1726.

### PR DESCRIPTION
**Pull request by hiddeco**
Fixes #1733 
Closes #1726

### CODE REVIEW

File: site/helm-integration.md [hunk 1]
```diff
@@ -230,6 +231,24 @@ setting `helmOperator.configureRepositories.enable` to `true` for the
...
 
+#### Azure ACR repositories
+
+For Azure ACR repositories, the entry in `repositories.yaml` created by
+running `az acr helm repo add` is unsufficient for the Helm operator.
+Instead you will need to [create a service principal](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-auth-service-principal#create-a-service-principal)
+and use the plain text id and password this gives you. For example:
+
+```yaml
+- caFile: ""
```
Comment (stefanprodan) [inline site/helm-integration.md]: I would add the stable repo too:
```yaml
- caFile: ""
  cache: stable-index.yaml
  certFile: ""
  keyFile: ""
  name: stable
  password: ""
  url: https://kubernetes-charts.storage.googleapis.com
```

Comment (hiddeco) [reply site/helm-integration.md]: Doesn't that distract from the actual information we are trying to get to the user?

Comment (stefanprodan) [reply site/helm-integration.md]: If they add only the ACR repo, any stable chart will fail. Maybe you could add a note on this and keep the yaml shot.

Comment (hiddeco) [reply site/helm-integration.md]: I think this note is indeed applicable to all users with a custom `repositories.yaml`, will add it to the top section.

**Pull request merged**